### PR TITLE
Truncate Post Body

### LINKED ISSUE #425: Truncate long posts instead of blocking them
Currently Helpy imposes a limit of 10,000 characters on `Posts`.  If the message is longer, it fails validation and cannot be created.  This is bad because if it is created via an email with a lot of quoted text, the `Post` will not be created, and fail silently.  Instead the better behavior would be to simply limit the character stored.

### PR DESCRIPTION
**Pull request by johnthepink**
This truncates the post body to 10,000 characters during validation. Fixes #425

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -113,4 +114,12 @@ def email_locale
     self.topic.locale.nil? ? I18n.locale : self.topic.locale.to_sym
   end
 
+  private
+
+    def truncate_body
```
Comment (johnthepink) [inline app/models/post.rb]: Totally. I'll get a fix going this evening. Thanks!

File: test/models/post_test.rb [hunk 2]
```diff
@@ -26,7 +26,6 @@ class PostTest < ActiveSupport::TestCase
   should validate_presence_of(:body)
   should validate_presence_of(:kind)
   should validate_presence_of(:user)
-  should validate_length_of(:body).is_at_most(10_000)
```
Comment (CGA1123) [inline test/models/post_test.rb]: Can this validation test not be kept?

File: app/models/post.rb (hunk 1 above)
Comment (lccezinha) [inline app/models/post.rb]: Thanks bro!

File: test/models/post_test.rb (hunk 2 above)
Comment (johnthepink) [inline test/models/post_test.rb]: @CGA1123 I intended to leave it in, but it fails now that we have the truncation. That test tries to set the body to 10,0001 "x"s:

```
Expected errors to include "is too long (maximum is 10000 characters)" when body is set to [insert 10,001 "x"]
```

Is there a way to bypass the `truncate_body` method for that test?

**Comment by johnthepink** [comment]
@lccezinha @CGA1123 pushed some updates

File: test/models/post_test.rb (hunk 2 above)
Comment (CGA1123) [inline test/models/post_test.rb]: @johnthepink 

Yeah, removing it is the only option I suppose, the new `Should truncate body length if greater than 10,000` test covers this validation anyways.

**Comment by CGA1123** [comment]
LGTM!

**Comment by lccezinha** [comment]
LGTM!

**Pull request merged**